src/xmidtsend_rbus.c

```diff
@@ -155,6 +156,15 @@ int checkCloudConn()
 			}
 		}
 	}
+	else if(get_close_retry())
+	{
+		ParodusInfo("cloud status is online but close_retry is in progress, wait till connection up\n");
+		sleep(5);
```

Comment:
When xmidt thread is idle and waiting, msgs expiry will happen, so thinking should we handle xmidtQOptmize() also in this close_retry enabled case ? 
If yes to do that, this "else if" condition can be moved as an OR condition inside "if" statement so that xmidtQOptmize and delete expired msgs also will be handled when retry is happening for longer duration.

Reply:
Else block from 152-155 is what is causing the problem as there is no sleep and leading to continuous retry. If we add the get_close_retry() as an OR its same as without this condition and the same problem will happen. I don't think queue optimise is needed as this close_retry wait is not going to be longer as the main thread will kick in and do retry to connect over new interface

Reply:
We cannot remove the else if block because then same issue as code flow will go to line 153 as cloud status is online